File: solutions/python/write.py

import splunklib.client as client
import utils
import sys
import datetime

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 3:
    print "Syntax: write.py index \"data\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
 
    index = service.indexes[sys.argv[1]]
    index.submit(str(datetime.datetime.now()) + ": " + sys.argv[2], 
                 host="myhost", source="mysource", sourcetype="mysourcetype")
    
except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: solutions/java/write.java
import com.splunk.*; 
import com.splunk.JobArgs.ExecutionMode;

import java.util.*;
import java.io.*;

public class write {

    public static void main(String[] args) {

		try {
			if(args.length != 2) {
				System.out.println("syntax: write index \"data\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);

			Service service = Service.connect(command.opts);

	        Args params = new Args();
		        params.put("sourcetype", "mysourcetype");
		        params.put("source", "mysource");
		        params.put("host", "myhost");
	        
	        Index index = service.getIndexes().get(args[0]);
	        index.submit(params, new Date() + ": " + args[1]);
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}